### ISSUE TITLE
Breadcrumb missing on read-only documents supplied by debug adapter

### ISSUE THREAD

**Issue by gjsjohnmurray** [Microsoft/vscode#62290]
Issue Type: <b>Bug</b>

VS Code version: Code 1.28.2 (7f3ce96ff4729c91352ae6def877e59c561f4850, 2018-10-17T00:23:51.859Z)
OS version: Windows_NT x64 10.0.15063

1. Start debugging something. In my case I am debugging a debug extension written in Typescript, e.g. the mock-debug sample.
2. Use the Pause button to suspend execution at a point where the debug adapter will supply the (read-only) source for display. In my case it was easy to suspend in async_hooks.js.
3. Observe that there's no breadcrumb on the loaded document. Toggling breadcrumb on the View menu has no effect.
4. Set a breakpoint in a source that exists on the local filesystem. Run to that breakpoint and confirm that a breadcrumb appears as expected. Toggling breadcrumb on the View menu works as expected.

Also occurs on current 1.29 Insider. I don't recall if the same problem existed in other breadcrumb-enabled versions (i.e. 1.26+).

**Comment by isidorn** [comment]
I can repro, breadcrumbs are not shown for content provided by the [`debugContentProvider`](https://github.com/Microsoft/vscode/blob/df8bba1247345d8c46daaaf87325dbaf64f21310/src/vs/workbench/parts/debug/browser/debugContentProvider.ts#L31)
Forwarding to @jrieken 
Is there something special I need to set for the breadcrumbs to be shwon for debug sources?

**Comment by jrieken** [comment]
> Is there something special I need to set for the breadcrumbs to be shwon for debug sources?

It depends on the scheme and if it's known by the file system providers - only then we can provide a meaningful path. Are you using virtual documents?

**Comment by isidorn** [comment]
I am not sure what is the definition of a virtual document but I am provding the content from memory. You can see a code pointer in my above comment on what I am doing, registering a contentProvider for the scheme `debug`. When an uri comes with that scheme I provide the content from memory.

**Comment by jrieken** [comment]
Yeah, those we ignore... Today, only schemes that are supported by the file system provider and the untitled scheme are supported, let's see what we can do here...

**Comment by mamilic** [comment]
Is this issue still active?
